### PR TITLE
Storage Blob Service Access Conditions, Bitwise enum class

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**

### CODE REVIEW

File: sdk/storage/inc/blobs/blob_options.hpp [hunk 1]
```diff
@@ -11,6 +12,65 @@
 
 namespace Azure { namespace Storage { namespace Blobs {
 
+  /**
+   * @brief Specifies access conditions for a container.
+   */
+  struct ContainerAccessConditions : public ModifiedTimeAccessConditions,
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_options.hpp:18]: How about `LastModifiedAccessCondition`, to align with returned value from response header.

File: sdk/storage/inc/blobs/blob_options.hpp [hunk 2]
```diff
@@ -11,6 +12,65 @@
...
+                                     public LeaseAccessConditions
+  {
+  };
+
+  /**
+   * @brief Specifies access conditions for a blob.
+   */
+  struct BlobAccessConditions : public ModifiedTimeAccessConditions,
+                                public ETagAccessConditions,
+                                public LeaseAccessConditions
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_options.hpp:28]: How about `LeaseIdAccessConditions`?

File: sdk/storage/test/append_blob_client_test.cpp [hunk 3]
```diff
@@ -57,19 +57,19 @@ namespace Azure { namespace Storage { namespace Test {
     EXPECT_EQ(properties.ContentLength, static_cast<int64_t>(m_blobContent.size()));
```
Comment (katmsft) [inline sdk/storage/test/append_blob_client_test.cpp:58]: TODO: Add test condition test cases.

Comment (JinmingHu-MSFT) [reply sdk/storage/test/append_blob_client_test.cpp:58]: https://github.com/Azure/azure-sdk-for-cpp/issues/261

File: sdk/storage/inc/blobs/blob_options.hpp (hunk 2 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_options.hpp:28]: talked offline

File: sdk/storage/inc/blobs/blob_options.hpp (hunk 1 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_options.hpp]: fixed

**Pull request merged**